app/services/time_tracker_extension/reports_sender/period.rb

```diff
@@ -0,0 +1,66 @@
+module TimeTrackerExtension
+  module ReportsSender
+    class Period
+      def initialize(period)
+        @period = period
+      end
+
+      def perform
+        return false if some_users_does_not_approve_period?
+        generate_reports
+        send_emails
+        remove_reports
+      end
+
+      private
+      attr_reader :period, :reports
+
+      def some_users_does_not_approve_period?
+        TimeTrackerExtension::TimeLockingPeriod.where(
+          beginning_of_period: period.beginning_of_period,
+          end_of_period: period.end_of_period,
+          workspace_id: period.workspace.id,
+          approved: false
+        ).any?
+      end
+
+      def generate_reports
+        @reports = users.inject({}) do |memo, user|
+          time_records_data = time_records_data(user)
+          report_generator = ReportGenerator.new(time_records_data, user)
+          memo[user.name] = report_generator.file
+          memo
+        end
+      end
+
+      def send_emails
```

Comment:
UserNotifier can do not only emails notifications, right? Then method name should be not only about emails.

Reply:
you're right, but we are send only emails with these time reports, so it's a double-edged sword. But yeah, let's rename that to `send_notifications`, because maybe we'll decide to send reports via Telegram as well